src/abstractarray.jl

```diff
@@ -204,6 +204,9 @@ end
 
 reshape(a::Array, ::Size{S}) where {S} = SizedArray{Tuple{S...}}(a)
 
+Base.rdims(out::Val{N}, inds::Tuple{Vararg{SOneTo}}) where {N} = Base.rdims(ntuple(i -> SOneTo(1), Val(N)), inds)
+Base.rdims(out::Tuple{Any}, inds::Tuple{Vararg{SOneTo}}) = (SOneTo(Base.rdims_trailing(inds...)),)
```

Comment:
```suggestion
Base.rdims(out::Val{N}, inds::Tuple{SOneTo, Vararg{SOneTo}}) where {N} = Base.rdims(ntuple(i -> SOneTo(1), Val(N)), inds)
Base.rdims(out::Tuple{Any}, inds::Tuple{SOneTo, Vararg{SOneTo}}) = (SOneTo(Base.rdims_trailing(inds...)),)
```
Overloading `rdims` seems fine but I think we should require that `inds` has at least one `SOneTo` to avoid type piracy.